pkgs/applications/science/misc/root/default.nix

```diff
@@ -53,6 +53,13 @@
 , CoreSymbolication
 , OpenGL
 , noSplash ? false
+  # To use the debug information on the fly (without installation)
+  # add the outPath of root.debug into NIX_DEBUG_INFO_DIRS (in PATH-like format)
+  # and make sure that gdb from Nixpkgs can be found in PATH.
+  #
+  # Building with debug info and separating them is not supported on all systems
+  # e.g. Darwin fails to support it (#203380)
+, withDebugInfo ? stdenv.hostPlatform.isLinux
```

Comment:
you don't need to make it optional because of darwin, setting separatedebuginfo to true is ignored on darwin